src/prefect/records/base.py

```diff
@@ -0,0 +1,152 @@
+import os
+import socket
+import threading
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Optional
+
+from prefect.results import BaseResult
+
+
+class RecordStore:
```

Comment:
Maybe use `abc` to signal what methods are abstract?

Reply:
I can do this in the next PR when all the signatures are aligned.